bot/exts/evergreen/youtube.py

```diff
@@ -0,0 +1,74 @@
+from typing import Dict, List, Optional
+
+from discord import Embed
+from discord.ext.commands import Bot, Cog, Context, command, cooldown
+from discord.utils import escape_markdown
+
+from bot.constants import Colours, Tokens
+
+KEY = Tokens.youtube
+SEARCH_API = "https://youtube.googleapis.com/youtube/v3/search?part=snippet&maxResults=5&q={search_term}&key={key}"
+YOUTUBE_URL = "https://www.youtube.com/watch?v={id}"
+RESULT = "`{index}` [{title}]({url}) - {author}"
+
+
+class YouTubeSearch(Cog):
+    """Sends the top 5 results of a query from YouTube."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+    async def search_youtube(self, search_term: str) -> Optional[List[Dict[str, str]]]:
```

Comment:
Why does this function return an `Optional`? Seems like it always returns a list.

Reply:
I would also say that a `DIct[str, str]` isn't a very accurate type -- maybe you could use `TypedDict` https://docs.python.org/3/library/typing.html#typing.TypedDict or make a special `NamedTuple` type?

Reply:
Yep, I'll remove `Optional`, and look into TypedDict, I agree that it wasn't the best typehint.